java/client/src/main/java/glide/managers/CommandManager.java

```diff
@@ -27,82 +18,53 @@ public class CommandManager {
     private final ChannelHandler channel;
 
     /**
-     * Async (non-blocking) get.<br>
-     * See <a href="https://redis.io/commands/get/">REDIS docs for GET</a>.
+     * Build a command and send.
      *
-     * @param key The key name
+     * @param command
+     * @param responseHandler - to handle the response object
+     * @return A result promise of type T
      */
-    public CompletableFuture<String> get(String key) {
-        return submitNewRequest(RequestType.GetString, List.of(key));
+    public <T> CompletableFuture<T> submitNewCommand(
+            Command command, RedisExceptionCheckedFunction<Response, T> responseHandler) {
+        // write command request to channel
+        // when complete, convert the response to our expected type T using the given responseHandler
+        return channel
+                .write(prepareRedisRequest(command.getRequestType(), command.getArguments()), true)
+                .thenApplyAsync(response -> responseHandler.apply(response));
     }
 
     /**
-     * Async (non-blocking) set.<br>
-     * See <a href="https://redis.io/commands/set/">REDIS docs for SET</a>.
+     * Build a protobuf command/transaction request object.<br>
+     * Used by {@link CommandManager}.
```

Comment:
Should this still have @param for this method's parameters (command and args)?

Reply:
Yes.  Thanks.